src/control/tools/EditSelection.cpp

```diff
@@ -25,7 +25,8 @@
 
 #define MINPIXSIZE 5  // smallest can scale down to, in pixels.
 
-EditSelection::EditSelection(UndoRedoHandler* undo, const PageRef& page, XojPageView* view) {
+EditSelection::EditSelection(UndoRedoHandler* undo, const PageRef& page, XojPageView* view):
+        SnapToGridInputHandler(view->getXournal()->getControl()->getSettings()) {
```

Comment:
Use composition over inheritance here, since this is a "has-an" not an "is-an" relationship.